org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -2534,6 +2534,19 @@ class CompletionOptions extends AbstractWorkDoneProgressOptions {
 	 */
 	List<String> triggerCharacters
 
+	/**
+	 * The list of all possible characters that commit a completion. This field
+	 * can be used if clients don't support individual commit characters per
+	 * completion item. See client capability
+	 * {@link CompletionItemCapabilities#commitCharactersSupport}.
+	 *
+	 * If a server provides both {@code allCommitCharacters} and commit characters on
+	 * an individual completion item the ones on the completion item win.
+	 *
+	 * Since 3.2.0
+	 */
+	List<String> allCommitCharacters
```

Comment:
Also needs to be added to `CompletionRegistrationOptions`. Another instance of the problem discussed in #505.

Reply:
This also involves what was discussed in #515 where it was pointed that that some *RegistrationOptions like `CompletionRegistrationOptions` are unused.

Reply:
Just a note that, although `CompletionRegistrationOptions` is not used as part of the return type for `completionProvider`, it is specified to be used for registration of the corresponding dynamic capability.